mkostemp is not available everywhere

On RHEL-5

During build:

```
 /builddir/build/BUILD/php-pecl-swoole-1.7.2/ZTS/src/network/TaskWorker.c:46: warning: implicit declaration of function 'mkostemp'
```

On PHP startup:

```
 PHP Warning:  PHP Startup: Unable to load dynamic library 'modules/swoole.so' - modules/swoole.so: undefined symbol: mkostemp in Unknown on line 0
```

--- a/config.m4
+++ b/config.m4
@@ -245,6 +245,7 @@ if test "$PHP_SWOOLE" != "no"; then
     AC_CHECK_LIB(c, accept4, AC_DEFINE(SW_USE_ACCEPT4, 1, [have accept4]))
     AC_CHECK_LIB(c, signalfd, AC_DEFINE(HAVE_SIGNALFD, 1, [have signalfd]))
     AC_CHECK_LIB(c, daemon, AC_DEFINE(HAVE_DAEMON, 1, [have daemon]))
+    AC_CHECK_LIB(c, mkostemp, AC_DEFINE(HAVE_MKOSTEMP, 1, [have mkostemp]))
     AC_CHECK_LIB(pthread, pthread_spin_lock, AC_DEFINE(HAVE_SPINLOCK, 1, [have pthread_spin_lock]))
     AC_CHECK_LIB(rt, clock_gettime, AC_DEFINE(HAVE_CLOCK_GETTIME, 1, [have clock_gettime]))
 

--- a/src/network/TaskWorker.c
+++ b/src/network/TaskWorker.c
@@ -43,7 +43,11 @@ int swTaskWorker_large_pack(swEventData *task, void *data, int data_len)
 {
 	swPackage_task pkg;
 	memcpy(pkg.tmpfile, SW_TASK_TMP_FILE, sizeof(SW_TASK_TMP_FILE));
+#ifdef HAVE_MKOSTEMP
 	int tpm_fd  = mkostemp(pkg.tmpfile, O_WRONLY);
+#else
+	int tpm_fd  = mkstemp(pkg.tmpfile);
+#endif
 	if (tpm_fd < 0)
 	{
 		swWarn("mkdtemp() failed. Error: %s[%d]", strerror(errno), errno);

